pkg/frontend/querymiddleware/split_and_cache.go

```diff
@@ -390,7 +401,11 @@ func (s *splitAndCacheMiddleware) storeCacheExtents(ctx context.Context, key str
 		return
 	}
 
-	s.cache.Store(ctx, map[string][]byte{cacheHashKey(key): buf}, resultsCacheTTL)
+	if lowerTTL {
+		s.cache.Store(ctx, map[string][]byte{cacheHashKey(key): buf}, resultsCacheLowerTTL)
+	} else {
+		s.cache.Store(ctx, map[string][]byte{cacheHashKey(key): buf}, resultsCacheTTL)
+	}
```

Comment:
I think that we should make the lower ttl apply only to the extent that is affected by it, otherwise a tenant with OOO enabled will have much higher load after this. Imagine someone rendering 30d graphs all the time, if they query up to now (which is the usual case), right now we cache the previous days for a week, but now we'll be recalculating them every 10 minutes.

I would add a `ValidUntil` timestamp to the `Extent`, and set it to `time.Now().Add(resultsCacheLowerTTL)` for the extents falling into the OOO window. Then when retrieving the cache, we would just filter out the ones that have `extent.ValidUntil > 0 && time.Now().After(extent.ValidUntil)`

Reply:
Can I use the same key to store different sets of extents? I don't totally know how this cache works, I will need to check. Also, a `ValidUntil` on the extent feels like duplicating what a cache already does.

Reply:
I didn't undertand this question, but I see your later changes and I think you're on the right path.

Reply:
Okay, clarified offline. I thought that all extents were stored under the same key, and that this change would expire multiple days every 30 min. 

Sorry for the misunderstanding. This looks good to me now.